docs/spec/relayer/Packets.md

```diff
@@ -0,0 +1,250 @@
+GetNextSequenceRecv(chain Chain, 
+                    portId Identifier, 
+                    channelId Identifier,  
+                    proofHeight Height) (uint64, CommitmentProof)
+
+// Returns packet acknowledgment with a commitment proof. If proof != nil, then it is being verified with the 
+// corresponding light client. Packet acknowledgment is queried at the given chain at the height proofHeight. 
+// If LATEST_HEIGHT is passed as a parameter, the query should be for latest height for which proof exists (MAX_HEIGHT - 1).
+GetPacketAcknowledgement(chain Chain, 
+                         portId Identifier, 
+                         channelId Identifier, 
+                         sequence uint64, 
+                         proofHeight Height) (bytes, CommitmentProof)
+ 
+// Returns estimate of the consensus height on the given chain. 
+GetConsensusHeight(chain Chain) Height
+
+// Returns estimate of the current time on the given chain. 
+GetCurrentTimestamp(chainB) uint64
+ 
+```
+
+## Computing destination chain
+
+```golang
+func GetDestinationInfo(ev IBCEvent, chainA Chain) Chain {
+    switch ev.type {
+        case SendPacketEvent: 
+            channel, proof = GetChannel(chainA, ev.sourcePort, ev.sourceChannel, ev.Height)
+            if proof == nil return nil
+                
+            connectionId = channel.connectionHops[0]
+            connection, proof = GetConnection(chainA, connectionId, ev.Height) 
+            if proof == nil return nil
+                
+            clientState = GetClientState(chainA, connection.clientIdentifier, ev.Height) 
+            return getHostInfo(clientStateA.chainID) 
+        ...    
+    }
+}
+```
+
+## Datagram creation logic
+
+### PacketRecv datagram creation
+
+```golang
+func createDatagram(ev SendPacketEvent, chainA Chain, chainB Chain, installedHeight Height) PacketRecv {        
+    
+    // Stage 1 
+    // Verify if packet is committed to chain A and it is still pending (commitment exists)
+    
+    proofHeight = installedHeight - 1
+    packetCommitment, packetCommitmentProof = 
+        GetPacketCommitment(chainA, ev.sourcePort, ev.sourceChannel, ev.sequence, proofHeight)     
+    if packetCommitmentProof != nil { return nil }
+        
+    if packetCommitment == null OR
+       packetCommitment != hash(concat(ev.data, ev.timeoutHeight, ev.timeoutTimestamp)) { return nil }
+            
+    // Stage 2 
+    // Execute checks IBC handler on chainB will execute
+    
+    channel, proof = GetChannel(chainB, ev.destPort, ev.destChannel, LATEST_HEIGHT)
+    if proof != nil { return nil }
+    
+    if channel == null OR
+       channel.state != OPEN OR
+       ev.sourcePort != channel.counterpartyPortIdentifier OR
+       ev.sourceChannel != channel.counterpartyChannelIdentifier { return nil }
+    
+    connectionId = channel.connectionHops[0]
+    connection, proof = GetConnection(chainB, connectionId, LATEST_HEIGHT) 
```

Comment:
It is a signal that the query should be made for the latest height. In Tendermint APIs it is normally achieved by putting 0 in case we are not interested in a particular height.